OneObjFunc.py

```diff
@@ -0,0 +1,327 @@
+
+        # TODO: better name for array
+        # TODO: Why is band here twice?
+        array, bgpsf = construct_psf_background(ra_grid, dec_grid,
+                                                cutout_wcs_list[i], x, y,
+                                                size,
+                                                roman_bandpasses[band],
+                                                color=0.61, psf=sim_psf,
+                                                pixel=pixel,
+                                                include_photonOps=False,
+                                                util_ref=util_ref,
+                                                use_roman=use_roman,
+                                                band=band)
+        # TODO comment this
+
+        if fit_background:
+            for j in range(num_total_imgs):
+                if i == j:
+                    bg = np.ones(size**2).reshape(-1, 1)
+                else:
+                    bg = np.zeros(size**2).reshape(-1, 1)
+                array = np.concatenate([array, bg], axis=1)
+
+        # Add the array of the model points and the background (if using)
+        # to the matrix of all components of the model.
+        psf_matrix.append(array)
+
+        # TODO make this not bad
+        if num_detect_imgs != 0 and i >= num_total_imgs - num_detect_imgs:
+            snx, sny = cutout_wcs_list[i].toImage(snra, sndec, units='deg')
+            if use_roman:
+                if use_real_images:
+                    pointing = exposures['Pointing'][i]
+                    SCA = exposures['SCA'][i]
+                else:
+                    pointing = 662
+                    SCA = 11
+                # sedlist is the length of the number of supernova
+                # detection images. Therefore, when we iterate onto the
+                # first supernova image, we want to be on the first element
+                # of sedlist. Therefore, we subtract by the number of
+                # predetection images: num_total_imgs - num_detect_imgs.
+                sn_index = i - (num_total_imgs - num_detect_imgs)
+                Lager.debug(f'Using SED #{sn_index}')
+                sed = sedlist[sn_index]
+                Lager.debug(f'x, y, snx, sny, {x, y, snx, sny}')
+                array = construct_psf_source(x, y, pointing, SCA,
+                                             stampsize=size, x_center=snx,
+                                             y_center=sny, sed=sed,
+                                             photOps=source_phot_ops)
+            else:
+                stamp = galsim.Image(size, size, wcs=cutout_wcs_list[i])
+                profile = galsim.DeltaFunction()*sed
+                profile = profile.withFlux(1, roman_bandpasses[band])
+                convolved = galsim.Convolve(profile, sim_psf)
+                array = convolved.drawImage(roman_bandpasses[band],
+                                            method='no_pixel',
```

Comment:
Consider making this hardcoded a keyword to the function with a default so that it's accessible, even if it's never used.